### DISCUSSION TITLE
Clans

### DISCUSSION THREAD

**Discussion by leolabdev** [Alt-Org/Altzone-WebPages#225] [Milestones]
Welcome to the Clans milestone discussion! Bring your ideas, questions, and suggestions here. Let’s collaborate to create a clans section that’s both vibrant and interactive, with all the features that will make it stand out

**Comment by Skoivumaki** [comment]
I decided to make this so its easier to follow development of Clans Mainpage. Stuff here is not final and is open for suggestions.
"issueID" will be replaced with the actual issue where its explained better.
##
Plan to implement new Clans Mainpage with features: Browse all clans, Clan leaderboard, My Clan and Store as coming soon page.

**Ive done my best to explain whats expected in the issues. If anything is unclear, please reach out to me on Discord for clarification!**

### TODO LIST:
_Issues marked with `*` need to be done before anything else and in order first to last! (with some exceptions)_

- [x] #298 *Flexbox div layout for planned components/containers in clans mainpage, both mobile and desktop
- [x] #299  *Navigation sidebar Component for Clan Mainpage between Clans, Leaderboard, My clan and Store, both mobile and desktop
- [x] #302  *Custom component for single clan with basic information. (name, logo, labels, members, score)
- [x] #303  *Add API leaderboard endpoint into clanApi
- [ ] #issueID Clans list with search, filters and pagination clans view, using the Custom component above (awaiting Figma design)
- [ ] #issueID Apply new visual look for Custom component (awaiting Figma design, same component as in #302)
- [ ] #407  Leaderboard page. We get list of clans from API using leaderboard and list them using the Custom component above
- [ ] #issueID Store page as coming soon page since store is still under development

### Potential issues:

> Since we are changing the layout to include a sidebar, this may break ClanRoomSubPage (clan overview) and ClanAddSubPage since they are not designed to work with a sidebar.
> So either we implement them to work with sidebar (change styling and include a back button on sidebar) or exclude them of the new layout. (use the old one with `[id]` and `addNew`) I dont know if this is possible/good practise, opinion? @leolabdev 
> Also if i remember correctly there was a sidebar that may be used?
> Edit: found it https://github.com/Alt-Org/Altzone-WebPages/discussions/230#discussioncomment-11097745 but it seems to be pretty basic so maybe we build on it?

Implemented a new sidebar that functions across the entire page and is compatible with the Clan page. #376

### Additional links/documentation
[Figma](https://www.figma.com/design/qaRrNbozqOeESL9lQ2UDrM/Altzone) Pages file Clans
[Google Drive](https://docs.google.com/presentation/d/1UHDjEPxilSh0wOhmLk_ywJ5yC372xF6Ivan3wNPlB0Q/edit?usp=sharing) Clans design plan (slides 10-13)
